frontend/packages/kubevirt-plugin/src/components/modals/cdrom-vm-modal/cdrom-modal.tsx

```diff
@@ -0,0 +1,221 @@
+import './_cdrom-modal.scss';
+
+export const CDRomModal = withHandlePromise((props: CDRomModalProps) => {
+  const {
+    vm,
+    handlePromise,
+    inProgress,
+    errorMessage,
+    persistentVolumeClaims,
+    storageClasses,
+    cancel,
+    close,
+  } = props;
+
+  const getCDsFromResource = (cds) =>
+    cds.map((disk) => {
+      const cd = {
+        ...initialDisk,
+        name: disk.name,
+        bootOrder: disk.bootOrder,
+        isInVM: true,
+      };
+      const container = getContainerImageByDisk(vm, cd.name);
+      if (container) {
+        if (_.includes(WINTOOLS_CONTAINER_NAMES, container))
+          return { ...cd, type: StorageType.WINTOOLS, windowsTools: container };
+        return { ...cd, type: StorageType.CONTAINER, container };
+      }
+
+      const url = getURLSourceByDisk(vm, cd.name);
+      if (url) {
+        const storageClass = getStorageClassNameByDisk(vm, cd.name);
+        const size = getStorageSizeByDisk(vm, cd.name).replace(/[^0-9]/g, '');
+        return { ...cd, type: StorageType.URL, url, storageClass, size };
+      }
+
+      return { ...cd, type: StorageType.PVC, pvc: getPVCSourceByDisk(vm, cd.name) };
+    });
+
+  const [cds, setCDs] = React.useState(getCDsFromResource(getCDRoms(vm)));
+  const [showRestartAlert, setShowRestartAlert] = React.useState(false);
+
+  const onCDChange = (cdName: string, key: string, value: string) => {
+    setCDs(
+      cds.filter((cd) => {
+        if (cdName === cd.name) {
+          cd.changed = true;
+          cd[key] = value;
+          setShowRestartAlert(true);
+
+          if (key === StorageType.URL) {
+            if (!value.startsWith('http://')) cd.isURLValid = false;
+            else cd.isURLValid = true;
+          }
+        }
+        return cd;
+      }),
+    );
+  };
+
+  const onCDAdd = () => {
+    let index = 1;
+    let name = `cd-drive-${index}`;
+    cds.forEach((cd) => {
+      if (cd.name === name) {
+        index++;
+        name = `cd-drive-${index}`;
+      }
+    });
+    const newCD = {
+      ...initialDisk,
+      type: StorageType.CONTAINER,
+      name,
+      isInVM: false,
+      changed: true,
+    };
+    setShowRestartAlert(true);
+    setCDs([...cds, newCD]);
+  };
+
+  const onCDEject = (cdName) =>
+    setCDs(
+      cds.filter((cd) => {
+        if (cdName === cd.name) {
+          cd.ejected = true;
+        }
+        return cd;
+      }),
+    );
+
+  const onCDDelete = (cdName) => {
+    const cdToDelete = cds.find((cd) => cd.name === cdName);
+    if (cdToDelete.isInVM) {
+      const patch = getRemoveDiskPatches(vm, { name: cdName });
+      const promise = k8sPatch(getVMLikeModel(vm), vm, patch);
+      // eslint-disable-next-line promise/catch-or-return
+      handlePromise(promise).then(() => {
+        setCDs(cds.filter((cd) => cd.name !== cdName));
+        setShowRestartAlert(true);
+      });
+    } else setCDs(cds.filter((cd) => cd.name !== cdName));
```

Comment:
AFAIK, we wrap all blocks, even one-liner, into `{ }`. I see this happening on other places within this PR as well.